src/store/reducers/ItemListReducer.ts

```diff
@@ -42,51 +45,53 @@ export const DELETE_ITEM = (id: string): Action => {
 type ListActions = ReturnType<typeof ADD_ITEM> | ReturnType<typeof DELETE_ITEM>;
 
 //reducer
-export const ItemListReducer: Reducer<IState, Action> = (state = initialState, action: ListActions) => {
-	switch (action.type) {
-		case `${ADD_ITEM}_SUCCESS`:
-			return {
-				...state,
-				items: [...state.items, action.payload]
-			}
-		case `${DELETE_ITEM}_SUCCESS`:
-			return {
-				...state,
-				items: state.items.filter((item, index) => index !== action.index)
-			}
-		default:
-			return state
-	}
-}
+export const ItemListReducer: Reducer<IState, Action> = (
+  state = initialState,
+  action: ListActions
+) => {
+  switch (action.type) {
+    // here's what i changed that made it work, my initial thought was that the name you used for these cases didn't match up
+    // with the action names you were dispatching. the code for updating the state was right
+    case ItemActions.ADD_ITEM:
+      return {
+        ...state,
+        items: [...state.items, action.payload]
+      };
+    case ItemActions.DELETE_ITEM:
+      return {
+        ...state,
+        items: state.items.filter((item, index) => index !== action.index)
+      };
+    default:
+      return state;
+  }
+};
 
 // Action Creators
 
+// here's where the next problem is, getting the new item data into the state
 export const addItem = (): Result<void> => {
-	let newItem: Item = {
-		id: v4(),
-		name: "",
-		cost: 0
-	}
-	return (dispatch, getState) => {
-		let items = getState().ItemList.items
-		const currentCopy = cloneDeep(items);
-
-		items = currentCopy.concat(newItem)
-
-		dispatch({ type: ItemActions.ADD_ITEM, items })
-	}
-}
+  let newItem: Item = {
+    id: v4(),
+    name: '',
+    cost: 0
+  };
+  console.log('test');
+  return (dispatch, getState) => {
+    // fixed it! was missing 'payload' lol
+    dispatch({ type: ItemActions.ADD_ITEM, payload: newItem });
```

Comment:
Holy! Wow thank you! 👏  👏 👏 
One can not simply assume a param name 🤣

Reply:
haha no problem :)